src/halmos/__main__.py

```diff
@@ -887,9 +700,18 @@ def run_sequential(run_args: RunArgs) -> List[TestResult]:
     return test_results
 
 
+def extend_args(args: Namespace, more_opts: str) -> Namespace:
+    if more_opts:
+        new_args = deepcopy(args)
```

Comment:
I was looking for something to avoid the need for a deepcopy here, I ended up with this:

```python
import argparse
import collections

parser = argparse.ArgumentParser()
parser.add_argument('--foo')
parser.add_argument('--bar')
parser.add_argument('--loop')

def dictify(args):
    return {k: v for k, v in vars(args).items() if v is not None}

defaults = parser.parse_args(['--foo', 'beep', '--bar', 'boop'])
overrides = parser.parse_args(['--foo', 'ding', '--loop', '42'])
combined = collections.ChainMap(dictify(overrides), dictify(defaults))

print(combined['foo']) # 'ding' comes from overrides, takes precedence over defaults
print(combined['bar']) # 'boop' comes from defaults
print(combined['loop']) # '42' comes from overrides, not specified in defaults
```

ChainMap is cool but in the end it's not necessarily better because there is still some copying involved because of a poor interaction with Namespace. Might make sense to write our own `NamespaceChain`-like wrapper that avoids any copying at all at a later point

Reply:
do you suggest using chainmap now, or doing something better later?

i wasn't super happy with using deepcopy here, but thought that the namespace object should be small enough and deepcopy wouldn't affect performance much. are there any other issues that i missed?

Reply:
no it's just a general dislike of deepcopy creep. Disregard for now